src/core/src/energy/functions.rs

```diff
@@ -104,6 +104,69 @@ impl PairPotential for LennardJones {
     }
 }
 
+/// Weeks Chandler Andersen Potential (WCA)
+///
+/// The following energy expression is used: `V(r) = 4 *
+/// epsilon * ((sigma/r)^12 - (sigma/r)^6) + epsilon` if `r < 2^1/6 * sigma
+/// 0 if r >= 2^1/6 * sigma`
+/// where `sigma` is the WCA
+/// distance constant, and `epsilon` the WCA energetic constant.
+/// # Examples
+///
+/// ```
+/// use lumol_core::energy::Potential;
+/// use lumol_core::energy::WeeksChandlerAndersen;
+///
+/// let potential = WeeksChandlerAndersen{sigma: 2.0, epsilon: 10.0};
+/// assert_eq!(potential.energy(2.0), 10.0);
+/// assert_eq!(potential.energy(3.0), 0.0);
+/// assert_eq!(potential.energy(1.0), 161290.0);
+///
+/// assert_eq!(potential.force(2.0), 120.0);
+/// ```
+
+#[derive(Clone, Copy)]
+pub struct WeeksChandlerAndersen {
+    /// Distance constant for the WeeksChandlerAndersen Potential
+    pub sigma: f64,
+    /// Energy Constant for the WeeksChandlerAndersen Potential
+    pub epsilon: f64,
+}
+
+impl Potential for WeeksChandlerAndersen {
+    fn energy(&self, r: f64) -> f64 {
+        if r < f64::powf(2., 1./6.) * self.sigma {
```

Comment:
For better efficiency, the `f64::powf(2., 1./6.) * self.sigma` value could be stored inside the `WeeksChandlerAndersen` struct, there is no need to recompute it every time. But that is not required.